### DISCUSSION TITLE
Thoughts on CONFIG_CHECKPOINT_RESTORE?

### DISCUSSION THREAD

**Discussion by ArthurKamalov** [kata-containers/kata-containers#10385] [General]
There is kind of an exotic use case, where long running and expensive processes that run inside kata containers are checkpointed by `criu`, so in case of some failure, their progress can be restored. For now, it seems that guest kernel is compiled without `CONFIG_CHECKPOINT_RESTORE` option and checkpoints are unavailable. 
I am not experienced it that and I'd venture a guess, that there are some obvious security considerations that I just cannot see, so I would grateful to hear an expert opinion on why it can or cannot be enabled. 

NOTE: Please do not confuse with "container runtime-level" checkpoint-restore. I am aware that it [is not supported ](https://github.com/kata-containers/kata-containers/blob/main/docs/Limitations.md#checkpoint-and-restore). I am asking about ability to checkpoint processes by using criu _inside_ a container.

Thanks in advance!

**Comment by ixcans** [comment]
Hi team,

Thanks for your work on this project. 

Following up on this, this feature has become increasingly important, especially with the rise of long-running workloads like AI agent / ML that depend on live migration. Without CONFIG_CHECKPOINT_RESTORE, it’s impossible to build reliable migration solutions on top.

Could you advise here? Should we build a image with `CONFIG_CHECKPOINT_RESTORE=y`?

cc: @stevenhorsman @fidencio

**Comment by fidencio** [comment]
@tz-torchai @ArthurKamalov, would you be able to raise a patch for us for this?

**Comment by ningmingxiao** [comment]
can kata-container use criu to checkpoint the workload container like runc checkpoint ? and I find clh https://intelkevinputnam.github.io/cloud-hypervisor-docs-HTML/docs/snapshot_restore.html  only support checkpoint  the vm, but user just need to checkpoint the application container to an oci image(will use less resources ), and then restore it by the checkpoint image , if it is needed  I can help  try to create a pr. @fidencio
